src/lib.rs

```diff
@@ -283,6 +285,38 @@ impl<B: Body> Body for http::Response<B> {
     }
 }
 
+impl Body for String {
+    type Data = VecDeque<u8>;
+    type Error = Infallible;
+
+    fn poll_data(
+        mut self: Pin<&mut Self>,
+        _cx: &mut Context<'_>,
+    ) -> Poll<Option<Result<Self::Data, Self::Error>>> {
+        if !self.is_empty() {
+            let s = std::mem::take(&mut *self);
+            Poll::Ready(Some(Ok(s.into_bytes().into())))
+        } else {
+            Poll::Ready(None)
+        }
```

Comment:
nit: my brain cannot deal with `!` + `else`

```suggestion
        if self.is_empty() {
            Poll::Ready(None)
        } else {
            let s = std::mem::take(&mut *self);
            Poll::Ready(Some(Ok(s.into_bytes().into())))
        }
```

Reply:
lol I kinda prefer the actual meat of the if statement to be first and keep it in order of how it will actually go (since you first return data then None).

Reply:
yeah makes sense too. I don't mind either way 😅